4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+#' @title Get site-specifc GLM predictions
+#' @description function to pull and combine GLM predictions for a single site
+#' for all 3 simulation periods based on the input meteo file and export file 
+#' vectors provided - may be pulling predictions for a single lake-gcm combo, 
+#' or predictions for all GCMS for a single lake.
+#' @param site_meteo_files the filepaths for the raw meteo data. The meteo data
+#' is used to determine the correct begin and end dates for the predictions and
+#' exclude predictions for any burn-in/burn-out periods.
+#' @param export_files the filepaths for the GLM predictions for the current lake
+#' for each time period and each GCM, if predictions for multiple GCMs are being 
+#' read in.
+#' @param run_gcm_list A list of GCM names that corresponds to the `site_meteo_files` 
+#' and `export_files` that are being read in. Depending on whether predictions are being
+#' read in for a single lake-gcm combo, or for all GCMs for a single lake, it may 
+#' be a list of the same GCM name repeated, or a list that includes all 6 GCM names.
+#' @return A munged dataframe of GLM predictions that includes columns for date, month, 
+#' year, DOY, decade, period, gcm, ice, hice (ice thickness), evap (evaporation),
+#' n_layers (number of simulated layers), and temperature predictions at 0.5-meter intervals
+get_site_preds <- function(site_meteo_files, site_export_files, run_gcm_list) {
```

Comment:
Seems like this function belongs somewhere in `3_extract`?

Reply:
Since it's site-specific I'm going to add it to the new `4_visualize/src/plot_data_utility_fxns.R` script